### PR TITLE
[TeamQuest] Define TeamQuest Level, add tests.

### PR DESCRIPTION
**Pull request by razvan3895**

### CODE REVIEW

File: wouso/games/teamquest/models.py [hunk 1]
```diff
@@ -12,11 +15,32 @@ def is_head(self):
             return False
         return self.group.head == self
 
+class TeamQuestLevel(models.Model):
+    questions = models.ManyToManyField(Question)
+    quest = models.ForeignKey('TeamQuest', null=True, blank=True, related_name='levels')
+    bonus = models.IntegerField(default=0)
+
+    @classmethod
+    def create(cls, quest, bonus, **kwargs):
```
Comment (iulianR) [inline wouso/games/teamquest/models.py]: Replace **kwargs with questions. You usually obtain a list of questions (e.g. by querying the db) and you want to pass that to the create method.

**Comment by iulianR** [comment]
We can have the tests failing for now. After we are done implementing the models, we delete the already added migrations and just redo the initial one. Any change to the model fields after that one should be accompanied by a migration.

File: wouso/games/teamquest/models.py [hunk 2]
```diff
@@ -13,10 +16,32 @@ def is_head(self):
         return self.group.head == self
```
Comment (razvand) [inline wouso/games/teamquest/models.py]: Use `owner` instead of head.

File: wouso/games/teamquest/models.py [hunk 3]
```diff
@@ -33,7 +58,6 @@ def __init__(self, *args, **kwargs):
         super(TeamQuestGame, self).__init__(*args, **kwargs)
 
 
-
 class TeamQuestGroup(PlayerGroup):
     head = models.OneToOneField('TeamQuestUser', null=True, blank=False)
```
Comment (razvand) [inline wouso/games/teamquest/models.py]: Rename `head` to `owner`.

File: wouso/games/teamquest/models.py [hunk 4]
```diff
@@ -13,10 +16,32 @@ def is_head(self):
         return self.group.head == self
 
 
+class TeamQuestLevel(models.Model):
+    questions = models.ManyToManyField(Question)
+    quest = models.ForeignKey('TeamQuest', null=True, blank=True, related_name='levels')
+    bonus = models.IntegerField(default=0)
```
Comment (razvand) [inline wouso/games/teamquest/models.py]: Add `blank` and `null` parameters.

File: wouso/games/teamquest/models.py [hunk 5]
```diff
@@ -13,10 +16,32 @@ def is_head(self):
...
+        return new_level
+
+    def add_question(self, question):
+        self.questions.add(question)
+
+    def remove_question(self, question):
+        if question in self.questions.all():
+            self.questions.remove(question)
+
+    def set_quest(self, quest):
```
Comment (razvand) [inline wouso/games/teamquest/models.py]: I would remove the method, as we would just use `level.quest = quest` in the implementation.

**Pull request merged**